### DISCUSSION TITLE
What's the purpose of SettledContract/ContractModification in contrast to ContractModification?

### DISCUSSION THREAD

**Discussion by renestreich** [OP-TED/eForms-SDK#120] [Q&A] [answered]
Hi everyone,

I started working with the eForms-SDK only recently, so forgive me if its purpose has been stated already anywhere else. I wonder why there is

/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:SettledContract/efac:ContractModification

in CANs while there is also

/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:ContractModification

What's the purpose of each of them? The example file can-modif_24_duration.xml in recent versions of the SDK describes the usage in the latter location while the same file in older versions of the SDK (< 0.3.0) has a ContractModification tied to the former location, so to individual SettledContracts. The former location also doesn't seem to have any schematron validations applied to it.

Could anyone elaborate on this one, please? :)

Thank you in advance
René

**Comment by YvesJo** [comment, marked as answer]
Hi Rene,
Thanks for sharing your question with us.
Indeed, it was originally planned to have Contract Modification information at Contract level; 
`/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:NoticeResult/efac:SettledContract/efac:ContractModification`
some discussions led to the conclusion that it should indeed be at root level. 
`/*/ext:UBLExtensions/ext:UBLExtension/ext:ExtensionContent/efext:EformsExtension/efac:ContractModification`
Rules` and documentation updates could not yet be finalized and will be integrated in a next release.

**Reply by renestreich** [reply]
Thank you for the clarification! I am looking forward to the next releases. :)
